TCMSApp/src/client/app/layout/sidebar.controller.js

```diff
@@ -1,28 +1,76 @@
-(function() {
+(function () {
     'use strict';
 
     angular
         .module('app.layout')
         .controller('SidebarController', SidebarController);
 
-    SidebarController.$inject = ['$state', 'routerHelper'];
+    SidebarController.$inject = ['$state', 'routerHelper', '$uibModal', 'moment', 'logger',
+        'sidebarFactory', 'authservice', 'createProjectFactory'];
     /* @ngInject */
-    function SidebarController($state, routerHelper) {
+    function SidebarController($state, routerHelper, $uibModal, moment, logger,
+                               sidebarFactory, authservice, createProjectFactory) {
         var vm = this;
-        var states = routerHelper.getStates();
 
-        activate();
+        vm.projectsNames = sidebarFactory.findProjectsNames();
+
+        vm.changeModal = function (modal) {
```

Comment:
do the `vm.changeModal` need to live in the scope of `SidebarController`?
If we use it only inside the main modal it can live there maybe.